src/game/level.c

```diff
@@ -711,8 +708,167 @@ bool Level_Load(int level_num)
+                case FT_DOOR:
+                case FT_ROOF:
+                case FT_TILT:
+                    k++;
+                    break;
+
+                case FT_LAVA:
+                    break;
+
+                case FT_TRIGGER: {
+                    uint16_t trig_type = (floor & 0x3F00) >> 8;
+                    uint16_t trigger_data = g_FloorData[k++];
+
+                    if (trig_type != TT_TRIGGER) {
+                        k++;
+                    }
+
+                    while (1) {
+                        int16_t command = g_FloorData[k++];
+                        int16_t item_idx = command & VALUE_BITS;
+                        if (trig_type == TT_TRIGGER) {
+                            ITEM_INFO *item = &g_Items[item_idx];
+                            OBJECT_INFO *object =
+                                &g_Objects[item->object_number];
+                            LOG_DEBUG(
+                                "item->object_number: %d", item->object_number);
+                            LOG_DEBUG("  item_idx: %d", item_idx);
+                            LOG_DEBUG(
+                                "  trigger_data: %d",
+                                trigger_data & IF_ONESHOT);
+                            LOG_DEBUG(
+                                "  Level_FindElement: %d",
+                                Level_FindElement(
+                                    item_idx, m_PierreIdxs, TT_SIZE));
+
+                            // Add Pierre pickup and kills if oneshot trigger
+                            if (item->object_number == O_PIERRE
+                                && trigger_data & IF_ONESHOT
+                                && !Level_FindElement(
+                                    item_idx, m_PierreIdxs, TT_SIZE)) {
+                                LOG_DEBUG("  Add Pierre #%d", item_idx);
+                                m_LevelPickups += PIERRE_ITEMS;
+                                m_LevelKillables += 1;
+                                m_PierreIdxs[pierre_idx++] = item_idx;
+                            }
+
+                            // Check for activated mummies
+                            if (item->object_number == O_MUMMY
+                                && !Level_FindElement(
+                                    item_idx, m_MummyIdxs, TT_SIZE)) {
+                                LOG_DEBUG("  Add Mummy #%d", item_idx);
+                                m_LevelKillables += 1;
+                                m_MummyIdxs[mummy_idx++] = item_idx;
+                            }
+
+                        } else {
+                            k++;
+                        }
+
+                        if (command & END_BIT) {
+                            break;
+                        }
+                    }
+                    break;
+                }
+                }
+
+                if (floor & END_BIT) {
+                    break;
+                }
+            }
+        }
+    }
+}
```

Comment:
Since this contains a lot of logic, let's have a new module for this, `levelstats` and move `getsecretcount` there as well.